it/src/test/scala/com/wavesplatform/it/sync/MassTransferSmartContractSuite.scala

```diff
@@ -0,0 +1,123 @@
+package com.wavesplatform.it.sync
+
+import com.wavesplatform.crypto
+import com.wavesplatform.it.api.SyncHttpApi._
+import com.wavesplatform.it.transactions.BaseTransactionSuite
+import com.wavesplatform.it.util._
+import com.wavesplatform.lang.v1.compiler.CompilerV1
+import com.wavesplatform.lang.v1.parser.Parser
+import com.wavesplatform.state._
+import com.wavesplatform.utils.dummyTypeCheckerContext
+import org.scalatest.CancelAfterFailure
+import play.api.libs.json.JsNumber
+import scorex.crypto.encode.Base58
+import scorex.transaction.Proofs
+import scorex.transaction.smart.SetScriptTransaction
+import scorex.transaction.smart.script.v1.ScriptV1
+import scorex.transaction.transfer.MassTransferTransaction.Transfer
+import scorex.transaction.transfer._
+import scala.concurrent.duration._
+
+/*
+Scenario:
+every month a foundation makes payments from two MassTransactions(type == 11):
+1) 80% to users
+2) 10% as tax and 10% to bank go after 30sec of payment from step 1)
+ */
+
+class MassTransferSmartContractSuite extends BaseTransactionSuite with CancelAfterFailure {
+  private val fourthAddress: String = sender.createAddress()
+  private val transferAmount: Long  = 1.waves
+  private val fee: Long             = 0.001.waves
+  private val massTransferFee       = 0.004.waves + 0.0005.waves * 4
+
+  test("airdrop emulation via MassTransfer") {
+    val scriptText = {
+      val untyped = Parser(s"""
+        let commonAmount = (tx.transfers[0].amount + tx.transfers[1].amount)
+        let totalAmountToUsers = commonAmount == 800000000
+        let totalAmountToGov = commonAmount == 200000000
+        let massTransferType = ((tx.type == 11) && (size(tx.transfers) == 2))
+
+        let accountPK = base58'${ByteStr(sender.publicKey.publicKey)}'
+        let accSig = sigVerify(tx.bodyBytes,tx.proofs[0],accountPK)
+
+        let txToUsers = (massTransferType && totalAmountToUsers)
+
+        let massTransferTx = getTransactionById(tx.proofs[1])
```

Comment:
It would be good to verify this transaction by `accountPK`, otherwise someone can set any mass transfer transaction in `tx.proofs[1]`